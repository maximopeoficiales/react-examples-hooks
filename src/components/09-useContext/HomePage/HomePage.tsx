import { useContext } from "react";
import { UserContext } from "../UserContext";

interface MyProps {}
const defaultProps: MyProps = {};
const HomePage = (props: MyProps) => {
  props = { ...defaultProps, ...props };

  // obtengo el contexto de UserContext
  const { user, setUser } = useContext<any>(UserContext);
  console.log(user);

  return (
    <div data-testid="HomePage" className="row justify-content-center">
      <div className="col-md-10">
        <h1>HomePage component</h1>
        <pre>{JSON.stringify(user, null, 3)}</pre>
      </div>
    </div>
  );
};

export default HomePage;
