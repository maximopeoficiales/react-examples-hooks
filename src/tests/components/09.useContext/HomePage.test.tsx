import { mount, shallow } from "enzyme";
import React from "react";
import HomePage from "../../../components/09-useContext/HomePage/HomePage";
import {
  User,
  UserContext,
} from "../../../components/09-useContext/UserContext";

describe("Pruebas en HomeScreen", () => {
  // mount monta el componente con todos sus hijos
  const user: User = {
    id: 1,
    age: 33,
    name: "jorge",
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser: () => {} }}>
      <HomePage />
    </UserContext.Provider>
  );

  test("debe de mostrarse correctament", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
