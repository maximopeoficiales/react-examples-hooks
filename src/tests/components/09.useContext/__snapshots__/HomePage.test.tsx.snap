// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomeScreen debe de mostrarse correctament 1`] = `
<div
  className="row justify-content-center"
  data-testid="HomePage"
>
  <div
    className="col-md-10"
  >
    <h1>
      HomePage component
    </h1>
    <pre>
      {
   "id": 1,
   "age": 33,
   "name": "jorge"
}
    </pre>
  </div>
</div>
`;
